### PR TITLE
Updates related to new plate colors

### PR DESCRIPTION
**Pull request by pjanik**
The ideas have been changing a lot, but I've updated this story description to reflect the latest requirements:
https://www.pivotaltracker.com/story/show/182121571

There's some UI work and a few changes to model logic related to plate IDs, colors, and merge and division.

I've also merged into this branch changes from this PR: https://github.com/concord-consortium/tectonic-explorer/pull/238, as they're touching the same code.

### CODE REVIEW

File: js/plates-model/model.ts [hunk 1]
```diff
@@ -77,10 +80,40 @@ export default class Model {
...
+      this.plates.forEach(p => {
+        availableHues.delete(p.hue);
+      });
+      return getRandomItem(availableHues);
+    }
+    // Very unlikely, but if all the available hues are used, return random color.
+    if (hueToAvoid !== undefined) {
+      availableHues.delete(hueToAvoid);
+    }
+    return getRandomItem(availableHues);
```
Comment (kswenson) [inline js/plates-model/model.ts:114]: Is random better than something like `availableHues[id % availableHues.length]`? Seems like there might be some value to a consistently repeating pattern.

**Review by kswenson — APPROVED** [review]
👍 Looks good!

File: js/plates-model/model.ts (hunk 1 above)
Comment (pjanik) [reply js/plates-model/model.ts:114]: This is the most unlikely use case when we already have 10 plates, and 11th is added (I don't think that's possible in practice with current settings).

File: js/components/boundary-config-dialog.tsx [hunk 2]
```diff
@@ -12,6 +12,7 @@ import BoundarySvg from "../../images/boundary.svg";
 import PlateArrow from "../../images/plate-arrow.svg";
 
 import css from "../../css-modules/boundary-config-dialog.less";
+import { plateHues } from "../plates-model/plate";
```
[the remark this answers is not in the document]
Comment (pjanik) [reply js/components/boundary-config-dialog.tsx]: Fixed.

**Pull request merged**